packages/rest/test/routes.test.ts

```diff
@@ -454,14 +454,59 @@ describe("rest: routes", () => {
       @test
       @autoRoute
       @sharedRoute
-      op get1(@path name: string): string;
+      op get1(@path foo: string): string;
 
       @test
       @autoRoute
-      op get2(@path name: string): string;
+      op get2(@path bar: string): string;
     `)) as { get1: Operation; get2: Operation };
 
     strictEqual(isSharedRoute(runner.program, get1), true);
     strictEqual(isSharedRoute(runner.program, get2), false);
   });
+
+  it("emits a diagnostic when @sharedRoute is used on action without explicit name", async () => {
+    const [_, diagnostics] = await compileOperations(
+      `
+      model Thing {
+        @key
+        @segment("things")
+        thingId: string;
+      }
+
+      @action
+      @autoRoute
+      @sharedRoute
+      op badAction(): {};
+
+      @action("good")
+      @autoRoute
+      @sharedRoute
+      op goodAction(): {};
+
+      @autoRoute
+      @sharedRoute
+      @collectionAction(Thing)
+      op badCollectionAction(): {};
+
+      @autoRoute
+      @sharedRoute
+      @collectionAction(Thing, "goodCollection")
+      op goodCollectionAction(): {};
+      `
+    );
+
+    expectDiagnostics(diagnostics, [
+      {
+        code: "@typespec/rest/shared-route-unspecified-action-name",
+        message:
+          "An operation marked as '@sharedRoute' must have an explicit collection action name passed to '@action'.",
+      },
+      {
```

Comment:
What role, if any, does `@actionSegment` play here? Do we need to account for it? Does this PR render it obselete?

It looks like it is still used. Should we make it a private decorator if it is only used internally (looks like collectionAction uses it).

Reply:
Looks like `@actionSegment` is already a private decorator, I think it's a helper that is only used internally for `@action` and `@collectionAction` to set the action name in the route string.